.github/workflows/notebook-test-suite.yml

```diff
@@ -0,0 +1,176 @@
+        run: |
+          if [ "${{ github.event.client_payload.harmony-environment || inputs.harmony-environment }}" == "uat" ]; then
+            echo "HARMONY_HOST_URL=https://harmony.uat.earthdata.nasa.gov"
+            echo "HARMONY_HOST_URL=https://harmony.uat.earthdata.nasa.gov" >> $GITHUB_ENV
+          elif [ "${{ github.event.client_payload.harmony-environment || inputs.harmony-environment }}" == "prod" ]; then
+            echo "HARMONY_HOST_URL=https://harmony.earthdata.nasa.gov"
+            echo "HARMONY_HOST_URL=https://harmony.earthdata.nasa.gov" >> $GITHUB_ENV
+          else
+            echo "HARMONY_HOST_URL=https://harmony.sit.earthdata.nasa.gov"
+            echo "HARMONY_HOST_URL=https://harmony.sit.earthdata.nasa.gov" >> $GITHUB_ENV
+          fi
+
+      - name: Create matrix configuration from Environment Variable
+        id: create-matrix
+        run: |
+          # Use the event type to select the appropriate service list
+          SERVICES_TESTS=$(cat ./config/services_tests_config_${{ github.event.client_payload.harmony-environment || inputs.harmony-environment }}.json | jq -r '."${{ github.event.action || inputs.service-name }}"')
+          # Split the selected service list into an array format and set it as the matrix
+          MATRIX=$(echo ",$SERVICES_TESTS" | tr ',' '\n' | awk '{$1=$1;print}' | jq -c -R '[inputs] | {service: .}')
+          echo "matrix=$MATRIX" >> $GITHUB_OUTPUT
+
+  test:
+    needs: base
+    environment: ${{ github.event.client_payload.harmony-environment || inputs.harmony-environment }}
+    runs-on: ubuntu-latest
+    outputs:
+      run_url: ${{ needs.base.outputs.run_url }}
+    strategy:
+      fail-fast: false
+      matrix: ${{ fromJson(needs.base.outputs.matrix) }}
+
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v4
+
+      - name: Pull Docker image
+        run: docker pull ghcr.io/nasa/regression-tests-${{ matrix.service }}:${{ github.event.client_payload.docker-image-tag || 'latest' }}
+
+      - name: Execute notebook
+        id: test-step
+        working-directory: ./test
+        env:
+          EDL_USER: ${{ secrets.EDL_USER }}
+          EDL_PASSWORD: ${{ secrets.EDL_PASSWORD }}
+          HARMONY_HOST_URL: ${{ needs.base.outputs.harmony_host_url }}
+        run: ./run_notebooks.sh ${{ matrix.service }}
+
+      - name: Save notebook as an artifact
+        if: always()
+        uses: actions/upload-artifact@v4
+        with:
+          name: notebook-output-${{ matrix.service }}
+          path: test/output/${{ matrix.service }}/Results.ipynb
+
+  email:
+    needs: test
+    if: always()
+    environment: ${{ github.event.client_payload.harmony-environment || inputs.harmony-environment }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Get TO_EMAIL
+        run: |
+          SERVICE_EMAILS=$(echo '${{ vars.SERVICES_EMAILS_CONFIG }}' | jq -r '."${{ github.event.action || inputs.service-name }}"')
```

Comment:
Without giving away anything sensitive, could you describe in a comment what the `SERVICES_EMAILS_CONFIG` looks like? (Or someone could add me as a repository admin so I can just go and take a look)

Reply:
Here is an example of the `SERVICES_EMAILS_CONFIG`:
```
{
  "harmony-service-example": "example@gmail.com",
  "hoss": "exmaple1@gmail.com,example@nasa.gov,example2@gmail.com"
}
```
There is also another configuration: `TO_EMAIL_DEFAULT` with value of a list of emails, that will be used if the service is not defined in the `SERVICES_EMAILS_CONFIG`.
Note: Currently NASA blocks the emails sent from my personal gmail account that I used for demo purpose. We will file a ticket and get the email with NASA email addresses working later.

Reply:
I have added you and Matt as collaborators in my fork. Hopefully, that will give you guys permission to see things.

Reply:
Thanks! That all makes sense.